### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by RokasMesk**
Different flows are being tested. I left the component tests too.

image

### CODE REVIEW

File: src/Aside/Aside.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { screen, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+import { renderWithProviders } from '../utils/testUtils';
+import Aside from './Aside';
+
+describe('Aside Component', () => {
+  describe('when initially rendered', () => {
+    it('should display the Create Event button', () => {
+      renderWithProviders(<Aside />);
+      expect(screen.getByText(/Create Event/)).toBeInTheDocument();
```
Comment (mantaspa) [inline src/Aside/Aside.spec.tsx]: try querying elements by their role when possible.
```suggestion
      expect(screen.getByRole('button', { name: 'Create Event' })).toBeInTheDocument();
```

File: src/Aside/Aside.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    });
+  });
+
+  describe('when the Create Event button is clicked', () => {
+    it('should dispatch the openModal action', () => {
+      const { store } = renderWithProviders(<Aside />);
+      fireEvent.click(screen.getByText(/Create Event/));
```
Comment (mantaspa) [inline src/Aside/Aside.spec.tsx]: same as above

File: src/Aside/MonthCalendar.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import MonthCalendar from './MonthCalendar';
+import { setCalendarDate } from '../store/slices/calendarDateSlice';
+
+describe('MonthCalendar Component', () => {
+  describe('when initially rendered', () => {
+    it('should display the correct month and year', () => {
+      const { store } = renderWithProviders(<MonthCalendar />);
+      const calendarDate = store.getState().calendar.calendarDate;
+      const monthYear = `${calendarDate.toLocaleString('default', { month: 'short' })} ${calendarDate.getFullYear()}`;
+      expect(screen.getByText(monthYear)).toBeInTheDocument();
```
Comment (mantaspa) [inline src/Aside/MonthCalendar.spec.tsx]:
```suggestion
      expect(screen.getByText('2024-01-01')).toBeInTheDocument();
```

Comment (mantaspa) [reply src/Aside/MonthCalendar.spec.tsx]: Try to be more specific with your selectors: https://testing-library.com/docs/dom-testing-library/api-within/

File: src/Aside/MonthCalendar.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    });
+    it('should display the MonthCalendar component', () => {
+      renderWithProviders(<MonthCalendar />);
+      const dayLabels = ['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa'];
+      dayLabels.forEach((label) => {
+        expect(screen.getByText(label)).toBeInTheDocument();
+      });
```
Comment (mantaspa) [inline src/Aside/MonthCalendar.spec.tsx]: Write dumb tests, no loops inside.
```suggestion
expect(screen.getByText('Sun')).toBeInTheDocument();
expect(screen.getByText('Mo')).toBeInTheDocument();
expect(screen.getByText('Tu')).toBeInTheDocument();
expect(screen.getByText('We')).toBeInTheDocument();
```

File: src/Aside/MonthCalendar.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      const newDate = new Date(2024, 1, 1);
+
+      store.dispatch(setCalendarDate(newDate));
+
+      renderWithProviders(<MonthCalendar />);
+
+      const updatedState = new Date(store.getState().calendar.calendarDate);
+      const monthYear = `${updatedState.toLocaleString('default', { month: 'short' })} ${updatedState.getFullYear()}`;
+      renderWithProviders(<MonthCalendar />);
+      expect(screen.getByText(monthYear)).toBeInTheDocument();
```
Comment (mantaspa) [inline src/Aside/MonthCalendar.spec.tsx]:
```suggestion
      expect(screen.getByText('2024 01 01')).toBeInTheDocument();
```

File: src/Aside/MonthCalendar.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        const newDate = new Date(2024, 1, 1);
+        store.dispatch(setCalendarDate(newDate));
+        fireEvent.click(screen.getByText('18'));
+        const updatedState = new Date(store.getState().calendar.calendarDate);
+        expect(updatedState.getDate()).toBe(18);
+      });
+      it('should update the calendar month when clicked on not current month date', () => {
+        const { store } = renderWithProviders(<MonthCalendar />);
+        const currentDate = store.getState().calendar.calendarDate;
+        const element = document.querySelector('.notCurrentMonthDay');
```
Comment (mantaspa) [inline src/Aside/MonthCalendar.spec.tsx]:
```suggestion
        const element = document.querySelector('.notCurrentMonthDay')!;
```

File: src/Aside/MonthCalendar.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+        if (element) {
+          fireEvent.click(element);
+        }
```
Comment (mantaspa) [inline src/Aside/MonthCalendar.spec.tsx]: no `ifs` in tests, let them fail
```suggestion
 
          fireEvent.click(element);
 
```

File: src/Header/Header.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+      renderWithProviders(<Header />);
+
+      expect(screen.getByText(/Calendar/)).toBeInTheDocument();
+    });
+
+    it('should have the initial state of calendar date', () => {
+      const { store } = renderWithProviders(<Header />);
+
+      const initialState = store.getState().calendar.calendarDate;
+      expect(initialState).toBeInstanceOf(Date);
```
Comment (mantaspa) [inline src/Header/Header.spec.tsx]: These tests should be testing the component, here you check only the redux state. Remove the test.
```suggestion
      const { store } = renderWithProviders(<Header />);

      const initialState = store.getState().calendar.calendarDate;
      expect(initialState).toBeInstanceOf(Date);
```

File: src/Header/Header.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    });
+  });
+
+  describe('when the date is changed via Redux action', () => {
+    it('should update the calendar date in the Redux state', () => {
```
Comment (mantaspa) [inline src/Header/Header.spec.tsx]: same, test the component trough UI, not via redux.

File: src/Header/Header.tsx [hunk 10]
```diff
@@ -53,7 +53,11 @@ const Header = () => {
 
 const TodayButton = ({ handleClick }: { handleClick: () => void }) => {
   return (
-    <button className={styles.headerButton} onClick={handleClick}>
+    <button
+      className={styles.headerButton}
+      onClick={handleClick}
+      aria-label="todayButton"
```
Comment (mantaspa) [inline src/Header/Header.tsx]: Why did you add these? These are meant for screen readers.
```suggestion
      aria-label="todayButton"
```

File: src/Modal/ModalContent.tsx [hunk 11]
```diff
@@ -130,7 +130,7 @@ export const CloseButton = ({ onClose }: CloseButtonProps) => {
 
 export const CreateEventButton = () => {
   return (
-    <button className={styles.eventSubmitButton} type="submit">
+    <button data-testid="addEventSubmit" className={styles.eventSubmitButton} type="submit">
```
Comment (mantaspa) [inline src/Modal/ModalContent.tsx]: query by role when possible
```suggestion
    <button className={styles.eventSubmitButton} type="submit">
```

File: src/flows/AddEventFlow.spec.tsx [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+      expect(addEventButton).toBeInTheDocument();
+
+      fireEvent.click(addEventButton);
+      expect(store.getState().modal.isModalOpen).toBe(true);
+
+      expect(screen.getByText(/Create new Event/i)).toBeInTheDocument();
+
+      const inputField: HTMLInputElement =
+        screen.getByLabelText(/Event Title/i);
+      fireEvent.change(inputField, { target: { value: 'Event 1' } });
```
Comment (mantaspa) [inline src/flows/AddEventFlow.spec.tsx]: replace fireEvent with userEvent where possible. In this case userEvent.type()

File: src/Header/Header.tsx (hunk 10 above)
Comment (RokasMesk) [reply src/Header/Header.tsx]: well I have to query it somehow, this button has no text, so I googled a solution to use aria-label. Could be data-testid but as I understand they are not preferable. I use it here: `userEvent.click(screen.getByRole('button', { name: /arrowRight/i }));`

Comment (mantaspa) [reply src/Header/Header.tsx]: Just use label `userEvent.click(screen.getByRole('button', { name: /Today/i }));`

and remove this one ;)

File: src/Header/Header.tsx [hunk 13]
```diff
@@ -73,6 +77,7 @@ const ArrowLeftButton = ({ handleClick }: { handleClick: () => void }) => {
     <button
       className={classNames(styles.headerButton, styles.noBorder)}
       onClick={handleClick}
+      aria-label="arrowLeft"
```
Comment (mantaspa) [inline src/Header/Header.tsx:80]: same here: `name: /Today/i`

Comment (mantaspa) [inline src/Header/Header.tsx]: Aria labels are announced by screen readers. Make it human friendly
```suggestion
      aria-label="Previous"
```

File: src/Header/Header.tsx [hunk 14]
```diff
@@ -84,6 +89,7 @@ const ArrowRightButton = ({ handleClick }: { handleClick: () => void }) => {
     <button
       className={classNames(styles.headerButton, styles.noBorder)}
       onClick={handleClick}
+      aria-label="arrowRight"
```
Comment (mantaspa) [inline src/Header/Header.tsx]:
```suggestion
      aria-label="Next"
```

File: src/WeekCalendar/WeekCalendar.spec.tsx [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+        events: [],
+        status: 'loading' as 'loading',
+        error: null,
+      },
+      calendar: {
+        calendarDate: new Date(),
+      },
+    };
+    renderWithProviders(<WeekCalendar />, { preloadedState });
+    expect(screen.getByTestId('loadingSpinner')).toBeInTheDocument();
```
Comment (mantaspa) [inline src/WeekCalendar/WeekCalendar.spec.tsx]: kebab-case
```suggestion
    expect(screen.getByTestId('loading-spinner')).toBeInTheDocument();
```
